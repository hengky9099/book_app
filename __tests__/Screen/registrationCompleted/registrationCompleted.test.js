import * as React from 'react';
import {create, act} from 'react-test-renderer';
import reduxTesting from '../../../src/helpers/reduxTesting';
import RegistrationCompleted from '../../../src/screens/registrationCompleted/index';

jest.mock('redux-persist', () => {
  const real = jest.requireActual('redux-persist');
  return {
    ...real,
    persistReducer: jest
      .fn()
      .mockImplementation((config, reducers) => reducers),
  };
});

jest.mock('@react-navigation/native');

const navigation = {
  navigate: jest.fn(),
};

describe('Registration Completed Testing', () => {
  const component = create(
    reduxTesting(<RegistrationCompleted navigation={navigation} />),
  );
  const root = component.root;

  describe('button to login', () => {
    test('navigate to login', () => {
      const button = component.root.findByProps({
        testID: 'RegistrationCompletedButton',
      }).props;

      act(() => button.onPress());
      expect(navigation.navigate).toBeCalledWith('Login');
    });
  });

  describe('should render', () => {
    test('default element', async () => {
      expect(root.props.store).toBeTruthy();
    });

    describe('Snapshot testing', () => {
      test('Snapshot', () => {
        const tree = create(<RegistrationCompleted />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
