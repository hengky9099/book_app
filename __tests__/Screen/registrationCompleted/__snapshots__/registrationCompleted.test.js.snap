// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Completed Testing should render Snapshot testing Snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "height": 1257.142857142857,
    }
  }
  testID="registrationCompleted"
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 37.714285714285715,
        "fontWeight": "bold",
        "left": 62.857142857142854,
        "top": 157.14285714285714,
      }
    }
  >
    Registration Completed!
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/Check.png",
      }
    }
    style={
      Object {
        "height": 314.2857142857143,
        "left": 110,
        "top": 314.2857142857143,
        "width": 314.2857142857143,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 31.428571428571427,
        "fontWeight": "800",
        "textAlign": "center",
        "top": 392.85714285714283,
      }
    }
  >
    We sent email verification to your email
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4D96FF",
        "borderRadius": 15.714285714285714,
        "height": 62.857142857142854,
        "justifyContent": "center",
        "left": 70.71428571428572,
        "opacity": 1,
        "top": 550,
        "width": 408.57142857142856,
      }
    }
    testID="RegistrationCompletedButton"
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      Back to Login
    </Text>
  </View>
</View>
`;
